common/test/acceptance/tests/studio/test_studio_outline.py

```diff
@@ -1315,8 +1315,12 @@ def test_expand_all_when_some_collapsed(self):
             And all sections are expanded
         """
         self.course_outline_page.visit()
+        # We have seen unexplainable sporadic failures in this test. Try disabling animations to see
+        # if that helps.
+        self.browser.execute_script("jQuery.fx.off = true;")
         self.course_outline_page.toggle_expand_collapse()
         self.assertEquals(self.course_outline_page.expand_collapse_link_state, ExpandCollapseLinkState.EXPAND)
+        self.verify_all_sections(collapsed=True)
         self.course_outline_page.section_at(0).expand_subsection()
```

Comment:
expand_subsection is where the failure actually was occurring, but the code looks good in there (using waits and all).